ggshield/core/git_hooks/ci/previous_commit.py

```diff
@@ -0,0 +1,141 @@
+import os
+from typing import Optional
+
+import click
+
+from ggshield.core.errors import UnexpectedError
+from ggshield.core.git_shell import get_last_commit_sha_of_branch
+from ggshield.core.utils import EMPTY_SHA
+
+from .supported_ci import SupportedCI
+
+
+def get_previous_commit_from_ci_env(
+    verbose: bool,
+) -> Optional[str]:
+    """
+    Returns the previous HEAD sha of the targeted branch.
+    Returns None if there was no commit before.
+    """
+    if os.getenv("GITLAB_CI"):
+        commit_sha = gitlab_previous_commit_sha(verbose)
+
+    elif os.getenv("GITHUB_ACTIONS"):
+        commit_sha = github_previous_commit_sha(verbose)
+
+    elif os.getenv("TRAVIS"):
+        raise NotImplementedError("Not implemented for Travis.")
+
+    elif os.getenv("JENKINS_HOME") or os.getenv("JENKINS_URL"):
+        raise NotImplementedError("Not implemented for Jenkins.")
+
+    elif os.getenv("CIRCLECI"):
+        raise NotImplementedError("Not implemented for CircleCI.")
+
+    elif os.getenv("BITBUCKET_COMMIT"):
+        raise NotImplementedError("Not implemented for BitBucket.")
+
+    elif os.getenv("DRONE"):
+        raise NotImplementedError("Not implemented for Drone.")
+
+    elif os.getenv("BUILD_BUILDID"):
+        raise NotImplementedError("Not implemented for Azure.")
+
+    else:
+        raise UnexpectedError(
+            f"Current CI is not detected or supported."
+            f" Supported CIs: {', '.join([ci.value for ci in SupportedCI])}."
+        )
```

Comment:
Nit: should we add all the non-supported CIs to this case instead of a NotImplementedError ?

Reply:
As I factorized all those `if` statements in the `SupportedCI` class, I can no longer do this.